kythe/go/extractors/config/wrapper/gradle_build_modifier_test.go

```diff
@@ -0,0 +1,123 @@
+package wrapper
+
+import (
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+)
+
+const testDataDir = "testdata"
+
+var multipleNewLines = regexp.MustCompile("\n{2,}")
+
+func filesEqual(got, want []byte) (bool, string) {
+	// remove superfluous whitespace
+	gotStr := strings.Trim(string(got[:]), " \n")
+	wantStr := strings.Trim(string(want[:]), " \n")
+	gotStr = multipleNewLines.ReplaceAllString(gotStr, "\n")
+	wantStr = multipleNewLines.ReplaceAllString(wantStr, "\n")
+
+	// diff want vs got
+	diff := cmp.Diff(gotStr, wantStr)
+	if diff != "" {
+		return false, diff
+	}
+
+	return true, ""
+}
+
+func TestHasKythe(t *testing.T) {
+	testcases := []struct {
+		f string
+		// Whether kythe javac wrapper should be present.
+		k bool
+		// Whether we expect an error.
+		e bool
+	}{
+		{"modified-gradle.build", true, false},
+		{"plain-gradle.build", false, false},
+		{"other-gradle.build", false, true},
+	}
+
+	for _, tcase := range testcases {
+		// This should just ignore the file and do nothing.
+		k, err := hasKytheWrapper(getPath(tcase.f))
+		if err != nil && !tcase.e {
+			t.Fatalf("Failed to open test gradle file: %v", err)
+		} else if err == nil && tcase.e {
+			t.Errorf("Expected a failure for file %s, but didn't get one.", tcase.f)
+		} else if tcase.k != k {
+			var errstr string
+			if tcase.k {
+				errstr = "should"
+			} else {
+				errstr = "should not"
+			}
+			t.Errorf("file %s %s already have kythe javac-wrapper", tcase.f, errstr)
+		}
+	}
+}
+
+func TestPreprocess(t *testing.T) {
+	testcases := []struct {
+		inputFile          string
+		expectedOutputFile string
+	}{
+		{"modified-gradle.build", "modified-gradle.build"},
+		{"plain-gradle.build", "modified-gradle.build"},
+	}
```

Comment:
I mean I guess I could add a case for the one that's supposed to fail and verify it forwards the error?